src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DoNotIgnoreReturnValue.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Runtime
+{
+    using static MicrosoftNetCoreAnalyzersResources;
+
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class DoNotIgnoreReturnValueAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string CA2022RuleId = "CA2022";
+
+        internal static readonly DiagnosticDescriptor DoNotIgnoreReturnValueRule = DiagnosticDescriptorHelper.Create(
+            CA2022RuleId,
+            CreateLocalizableResourceString(nameof(DoNotIgnoreReturnValueTitle)),
+            CreateLocalizableResourceString(nameof(DoNotIgnoreReturnValueMessage)),
+            DiagnosticCategory.Reliability,
+            RuleLevel.IdeSuggestion,
+            CreateLocalizableResourceString(nameof(DoNotIgnoreReturnValueDescription)),
+            isPortedFxCopRule: false,
+            isDataflowRule: false);
+
+        internal static readonly DiagnosticDescriptor DoNotIgnoreReturnValueRuleWithMessage = DiagnosticDescriptorHelper.Create(
+            CA2022RuleId,
+            CreateLocalizableResourceString(nameof(DoNotIgnoreReturnValueTitle)),
+            CreateLocalizableResourceString(nameof(DoNotIgnoreReturnValueMessageCustom)),
+            DiagnosticCategory.Reliability,
+            RuleLevel.IdeSuggestion,
+            CreateLocalizableResourceString(nameof(DoNotIgnoreReturnValueDescription)),
+            isPortedFxCopRule: false,
+            isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(DoNotIgnoreReturnValueRule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterOperationBlockStartAction(context =>
+            {
+                if (!context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemDiagnosticsCodeAnalysisDoNotIgnoreAttribute, out var doNotIgnoreAttribute))
+                {
+                    return;
+                }
+
+                context.RegisterOperationAction(context =>
+                {
+                    var invocation = (IInvocationOperation)context.Operation;
+
+                    if (!invocation.TargetMethod.ReturnsVoid && invocation.Parent.Kind == OperationKind.ExpressionStatement)
```

Comment:
The API review discussion also talks about the attribute applying to the `TResult` in the case of the return type being a `Task<TResult>`, e.g. that if you had:
```C#
[return: DoNotIgnore]
public Task<TResult> CreateAsync();
```
you'd not only get a warning for:
```C#
CreateAsync();
```
but also for:
```C#
await CreateAsync();
```
since even though you didn't ignore the `Task<TResult>`, you did ignore the `TResult`.  Is that still something we plan to do?

Reply:
Oh gosh, I overlooked that in the description. Thanks; I'll work to add support for that.

Reply:
`Task<TResult>` support has been added in.